File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int cnt=0;
        int sz = nums.size();
        vector<int> low, high;
        for(int i : nums){
            if(i < pivot) low.push_back(i);
            else if(i > pivot) high.push_back(i);
            else cnt++;
        }

        for(int i=0; i<cnt; i++) low.push_back(pivot);

        low.insert(low.end(), high.begin(), high.end());
        return low;
    }
};

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int sz = nums.length, k1=0, k2=0;
        int[] ans = new int[sz];

        for(int i=0; i<sz; i++){
            if(nums[i] < pivot){
                ans[k1++] = nums[i];
            }
            else if(nums[i] == pivot) k2++;
        }

        for(int i=0; i<k2; i++){
            ans[k1++] = pivot;
        }

        for(int i=0; i<sz; i++){
            if(nums[i] > pivot) ans[k1++] = nums[i];
        }

        return ans;
    }
}